bolt-modules/out/lib/puppet/functions/out/message.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+Puppet::Functions.create_function(:'out::message') do
+  COLORS = %w[red green yellow].freeze
+
+  dispatch :output_message do
+    param 'String', :message
+    optional_param 'Hash[String[1], Any]', :options
+  end
+
+  def output_message(message, options = {})
+    executor = Puppet.lookup(:bolt_executor) { nil }
```

Comment:
Instead of ignoring `out::message` in an apply block should we instead raise an error like the other functions that are not allowed in an apply block? 
```
    unless Puppet[:tasks]
      raise Puppet::ParseErrorWithIssue
        .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'out::message')
    end
```

Reply:
Good call, will do